[BE] S3 구현체 등록 문제

### 버그 요약

S3 구현체 등록에 실패 했습니다. 

### 발생 위치

S3

### 에러 및 캡처

2026-01-29 05:35:28.692 ERROR [      main] pid=657062 o.s.b.d.LoggingFailureAnalysisReporter   : 
***************************
APPLICATION FAILED TO START
***************************
Description:
Parameter 1 of constructor in com.tasteam.infra.storage.s3.S3StorageClient required a bean of type 'com.amazonaws.auth.AWSCredentialsProvider' that could not be found.
Action:
Consider defining a bean of type 'com.amazonaws.auth.AWSCredentialsProvider' in your configuration.
=== 배포 실패 ===
기존 서버(GREEN)는 계속 실행 중입니다.
전체 로그: /home/***/backend/app-20260129_053516.log
2026/01/29 05:35:32 Process exited with status 1
Error: Process completed with exit code 1.

### 재현 절차

_No response_

### 예상 동작

_No response_

### 참고사항

_No response_

--- a/app-api/src/main/java/com/tasteam/infra/storage/s3/S3StorageConfig.java
+++ b/app-api/src/main/java/com/tasteam/infra/storage/s3/S3StorageConfig.java
@@ -6,8 +6,10 @@ import org.springframework.context.annotation.Configuration;
 import org.springframework.context.annotation.Profile;
 import org.springframework.util.Assert;
 
+import com.amazonaws.auth.AWSCredentialsProvider;
 import com.amazonaws.auth.AWSStaticCredentialsProvider;
 import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.auth.DefaultAWSCredentialsProviderChain;
 import com.amazonaws.services.s3.AmazonS3;
 import com.amazonaws.services.s3.AmazonS3ClientBuilder;
 import com.tasteam.infra.storage.StorageProperties;
@@ -18,17 +20,21 @@ import com.tasteam.infra.storage.StorageProperties;
 public class S3StorageConfig {
 
 	@Bean
-	public AmazonS3 amazonS3(StorageProperties properties) {
+	public AWSCredentialsProvider awsCredentialsProvider(StorageProperties properties) {
+		if (properties.hasStaticCredentials()) {
+			return new AWSStaticCredentialsProvider(
+				new BasicAWSCredentials(properties.getAccessKey(), properties.getSecretKey()));
+		}
+		return DefaultAWSCredentialsProviderChain.getInstance();
+	}
+
+	@Bean
+	public AmazonS3 amazonS3(StorageProperties properties, AWSCredentialsProvider credentialsProvider) {
 		Assert.hasText(properties.getRegion(), "tasteam.storage.region은 필수입니다");
 
-		AmazonS3ClientBuilder builder = AmazonS3ClientBuilder.standard()
-			.withRegion(properties.getRegion());
-
-		if (properties.hasStaticCredentials()) {
-			builder.withCredentials(new AWSStaticCredentialsProvider(
-				new BasicAWSCredentials(properties.getAccessKey(), properties.getSecretKey())));
-		}
-
-		return builder.build();
+		return AmazonS3ClientBuilder.standard()
+			.withRegion(properties.getRegion())
+			.withCredentials(credentialsProvider)
+			.build();
 	}
 }

--- a/app-api/src/main/resources/application.dev.yml
+++ b/app-api/src/main/resources/application.dev.yml
@@ -69,6 +69,8 @@ naver:
     api-key: ${NAVER_MAPS_API_KEY:dummy}
 
 tasteam:
+  storage:
+    type: s3
   webhook:
     enabled: true
     provider: discord

--- a/app-api/src/main/resources/application.prod.yml
+++ b/app-api/src/main/resources/application.prod.yml
@@ -69,6 +69,7 @@ spring:
 
 tasteam:
   storage:
+    type: s3
     region: ${STORAGE_REGION:}
     bucket: ${STORAGE_BUCKET:}
     base-url: ${STORAGE_BASE_URL:}

